pkg/apis/provisioning/v1alpha5/provisioner.go

```diff
@@ -132,6 +134,12 @@ type ProvisionerList struct {
 // by their priority weight in-place
 func (pl *ProvisionerList) OrderByWeight() {
 	sort.Slice(pl.Items, func(a, b int) bool {
-		return pl.Items[a].Spec.Weight > pl.Items[b].Spec.Weight
+		if pl.Items[b].Spec.Weight == nil {
```

Comment:
Can `ptr.Int32(pl.Items[b].Spec.Weight)` simplify these if checks? It converts nil -> 0, which is the semantic we want.